models/knownorigin/knownorigin_ethereum_events.sql

```diff
@@ -0,0 +1,125 @@
+                                "project",
+                                "knownorigin",
+                                \'["cat"]\') }}'
+    )
+}}
+
+with raw_ko_transactions as (
+    select  a.evt_block_time
+            , b.contract_address
+            , _tokenId as token_id
+            , _seller as seller 
+            , _buyer as buyer
+            , _price as raw_amount
+            , a.evt_block_number
+            , a.evt_index
+            , a.evt_tx_hash
+            , case when b.from = '0x0000000000000000000000000000000000000000' then 'Primary' else 'Secondary' end as sale_type
+            , 'Buy' as trade_category
+            , lower('0xc322cdd03f34b6d25633c2abbc8716a058c7fe9e') as project_contract_address
+    from {{ source('knownorigin_ethereum','TokenMarketplaceV2_evt_TokenPurchased') }} a
+    inner join {{ source('erc721_ethereum','evt_transfer') }} b on a.evt_tx_hash = b.evt_tx_hash and a._tokenId = b.tokenId
+    {% if is_incremental() %}
+    where a.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+
+    union all 
+    
+    select a.evt_block_time
+            , b.contract_address
+            , _tokenId as token_id
+            , _currentOwner as seller
+            , _bidder as buyer
+            , _amount as raw_amount
+            , a.evt_block_number
+            , a.evt_index
+            , a.evt_tx_hash
+            , case when b.from = '0x0000000000000000000000000000000000000000' then 'Primary' else 'Secondary' end as sale_type
+            , 'Offer Accepted' as trade_category
+            , lower('0xc1697d340807324200e26e4617ce9c0070488e23') as project_contract_address
+    from {{ source('knownorigin_ethereum','TokenMarketplace_evt_BidAccepted') }} a
+    inner join {{ source('erc721_ethereum','evt_transfer') }} b on a.evt_tx_hash = b.evt_tx_hash and a._tokenId = b.tokenId
+    {% if is_incremental() %}
+    where a.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+
+    union all 
+    
+    select a.evt_block_time
+            , a.contract_address
+            , _tokenId as token_id
+            , b.from as seller
+            , _buyer as buyer
+            , _priceInWei as raw_amount
+            , a.evt_block_number
+            , a.evt_index
+            , a.evt_tx_hash
+            , case when b.from = '0x0000000000000000000000000000000000000000' then 'Primary' else 'Secondary' end as sale_type
+            , 'Buy' as trade_category
+            , lower('0xfbeef911dc5821886e1dda71586d90ed28174b7d') as project_contract_address    
+    from {{ source('knownorigin_ethereum','KnownOriginDigitalAssetV2_evt_Purchase') }} a
+    inner join {{ source('erc721_ethereum','evt_transfer') }} b on a.evt_tx_hash = b.evt_tx_hash and a._tokenId = b.tokenId
+    {% if is_incremental() %}
+    where a.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+
+)
+
+select  'ethereum' as blockchain
```

Comment:
can you explicitly cast columns that are done so [here](https://github.com/duneanalytics/spellbook/blob/main/models/archipelago/ethereum/archipelago_ethereum_events.sql)?

Reply:
we want to be consistent across nft sector event tables